ranch/storage.go

```diff
@@ -418,11 +423,25 @@ func (s *Storage) syncDynamicResourceLifeCycles(newDRLCByType, existingDRLCByTyp
 			if !reflect.DeepEqual(existingDRLC, newDRLC) {
 				dRLCToUpdate = append(dRLCToUpdate, newDRLC)
 			}
-		} else {
+		} else if existingDRLC.Spec.MinCount != 0 || existingDRLC.Spec.MaxCount != 0 {
 			// Mark for deletion of all associated dynamic resources.
 			existingDRLC.Spec.MinCount = 0
 			existingDRLC.Spec.MaxCount = 0
-			dRLCToUpdate = append(dRLCToUpdate, existingDRLC)
+			if err := s.client.Update(s.ctx, existingDRLC.DeepCopyObject()); err != nil {
```

Comment:
Why is it necessary to do this update in the loop?

(Partially I'm wondering why we treat this as a special update, rather than just passing the DRLC to deleteDynamicResourceLifecycles, though I guess we don't have the list of static resources here. Hm.)

I guess it feels like this code is a bit messy and needs some cleaning up, though that does predate this PR. Thoughts?

Reply:
it needs to be in the loop because only for this change we must make sure to again have it in the cache before proceeding.

Reply:
> I guess it feels like this code is a bit messy and needs some cleaning up, though that does predate this PR. Thoughts?

Certainly, for example the approach here of constructing a slice to then pass that to another func that serially updates all elements in the slice is just needlessly complex and adds a couple of allocations, just directly updating here would be simpler. But The change already has an about 400 line diff, so I wanted to not make that worse